### PR TITLE
[fix] set full to false in Reset()

### PR DESCRIPTION
**Pull request by neal-zhu**
Sorry about forgetting to reset ``q.full``

### CODE REVIEW

File: queue/bytes_queue.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ func (q *BytesQueue) Reset() {
 	q.head = leftMarginIndex
 	q.rightMargin = leftMarginIndex
 	q.count = 0
+	q.full = false
```
Comment (janisz) [inline queue/bytes_queue.go]: Can you add a test for this?

Comment (neal-zhu) [reply queue/bytes_queue.go]: I'd like to, but what kind of test do you mean? 
I think it's kind of obvious that without resetting this flag, the queue will still be full after ``Reset()``, and the next ``Push()`` will cause a re-allocation on this empty queue

Comment (janisz) [reply queue/bytes_queue.go]: Exactly,

**Comment by neal-zhu** [comment]
Reopen a new one

**Comment by janisz** [comment]
https://github.com/allegro/bigcache/pull/213

**Pull request closed without merging**